# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../ccache/dont_stat_compiler.patch
# Copyright (C) 2004 - 2010 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

For us the t2 wrapper is the compiler executable which is often rebuild (e.g.
on config change. So we most often want to prevent hashing the compiler (or
better our wrapper) size and date ...

(Originally written in ROCK Linux, commented and rediffed for recent
 ccache versions.)

  - Rene Rebe <rene@exactcode.de>

diff -Nur ccache-3.0-orig/ccache.c ccache-3.0/ccache.c
--- ccache-3.0-orig/ccache.c	2010-07-13 10:33:43.962912291 +0000
+++ ccache-3.0/ccache.c	2010-07-13 10:38:23.602913263 +0000
@@ -913,7 +913,8 @@
 	hash_delimiter(hash, "ext");
 	hash_string(hash, i_extension);
 
-	if (stat(args->argv[0], &st) != 0) {
+        if (!getenv("CCACHE_DONT_STAT_COMPILER")) {
+        if (stat(args->argv[0], &st) != 0) {
 		cc_log("Couldn't stat the compiler (%s)", args->argv[0]);
 		stats_update(STATS_COMPILER);
 		failed();
@@ -936,6 +937,7 @@
 		hash_int(hash, st.st_size);
 		hash_int(hash, st.st_mtime);
 	}
+	}
 
 	/*
 	 * Also hash the compiler name as some compilers use hard links and
